include/beast/core/detail/sha1.hpp

```diff
@@ -275,15 +275,14 @@ finish(sha1_context& ctx, void* digest) noexcept
     std::uint64_t total_bits =
         (ctx.blocks*64 + ctx.buflen) * 8;
     // pad
-    auto const buflen = ctx.buflen;
     ctx.buf[ctx.buflen++] = 0x80;
+    auto const buflen = ctx.buflen;
```

Comment:
I am very reluctant to change this code since it is based on another public work and it would be nice to make as few changes as possible (except those necessary for C++).